clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -561,12 +562,25 @@ private static int parseAcks(String acksString) {
      * @throws org.apache.kafka.common.errors.AuthorizationException fatal error indicating that the configured
      *         transactional.id is not authorized. See the exception for more details
      * @throws KafkaException if the producer has encountered a previous fatal error or for any other unexpected error
+     * @throws TimeoutException if the time taken for initialize the transaction has surpassed <code>max.block.ms</code>.
```

Comment:
Can you add a comment to the javadoc mentioning that this method may be retried if a `TimeoutException` or an `InterruptException` is raised?